### PR TITLE
MixerSession connect is no longer async, and update MediaStream to work in node context

### PR DESCRIPTION
**Pull request by hd23408**
To fix tests!

### CODE REVIEW

File: src/classes/HiFiMixerSession.ts [hunk 1]
```diff
@@ -639,12 +653,15 @@ export class HiFiMixerSession {
...
             this._onConnectionStateChange(HiFiConnectionStates.Failed, { success: false, error: error });
+            return false;
         })
         .finally(() => {
             this._raviSignalingConnection.removeStateChangeHandler(tempUnavailableStateHandler);
             this.concurrency = 0;
             this._tryingToConnect = false;
         });
+
+        return true;
```
Comment (howard-stearns) [inline src/classes/HiFiMixerSession.ts:664]: I guess the promise itself is evaluated for effect, and the value not used anywhere? 
So it's intended to return true, and not the promise?

Comment (howard-stearns) [reply src/classes/HiFiMixerSession.ts:664]: And the doc string says it returns a boolean indicating success or failure, but we don't actually know the success at this point in time, right?

Comment (hd23408) [reply src/classes/HiFiMixerSession.ts]: This is why I needed more eyes on it! :-) Thank you, @howard-stearns .  I've moved the `return true` inside the last `then`, and at that point we SHOULD know that it was successful. So I believe that with this _latest_ change, it is appropriate for `HiFiMixerSession.connectToHiFiMixer()` to be considered synchronous, and the boolean it returns should, in fact, indicate success or failure. (I think.) Does that sound right?

Comment (hd23408) [reply src/classes/HiFiMixerSession.ts]: @howard-stearns Thank you again so much for your help with this! `connectToHiFiMixer()` now returns `void` with an explicit note about getting the "return value" ONLY through the use of callbacks. The test has been updated to use a callback and check to make sure it gets called with the appropriate values. Whew.

File: tests/integration/serverConnections.integration.test.ts [hunk 2]
```diff
@@ -127,6 +127,7 @@ describe('Mixer connections', () => {
     describe('Connections that should be denied', () => {
         beforeEach(() => {
             hifiCommunicator = new HiFiCommunicator();
+            jest.setTimeout(30000);
```
Comment (hd23408) [inline tests/integration/serverConnections.integration.test.ts:130]: These timeouts allow time for auto-reconnects to finish up when we're defaulting auto-reconnects to "true"

**Review by howard-stearns — APPROVED** [review]

**Pull request merged**